activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -103,6 +103,16 @@ def index_name_for_remove(table_name, options = {})
       end
 
       class V5_0 < V5_1
+        def create_table(table_name, options = {})
+          # Since version 5.1 Postgres adapter uses bigserial type for primary
+          # keys by default. This compat layer makes old migrations utilize
+          # serial type instead, the way it used to work before 5.1
+          if connection.adapter_name == "PostgreSQL"
```

Comment:
This needs to handle MySQL as well. Is it possible for us to implement this in an adapter agnostic way?